configs/projects/puppet-agent.rb

```diff
@@ -241,7 +241,12 @@
   end
   proj.component "ruby-shadow" unless platform.is_aix? || platform.is_windows?
   proj.component "ruby-augeas" unless platform.is_windows?
-  proj.component "openssl"
+
+  fips_build = ENV["USE_SYSTEM_OPENSSL"]
+  if fips_build != '1'
+    proj.component "openssl"
+  end
```

Comment:
Hi Jayant. Since `ENV` is treated as a Ruby Hash, calling `ENV["USE_SYSTEM_OPENSSL"]` will return either `nil` ([which is `false`, along with `false`](http://www.skorks.com/2009/09/true-false-and-nil-objects-in-ruby/)) or any other object, which will then be treated as `true`. So you can simplify almost all of these conditionals:

```ruby
proj.component "openssl" unless ENV["USE_SYSTEM_OPENSSL"]
```

You could also use this to set a project-wide setting for OpenSSL so you can skip the Env. checks in all of the components and use the `Project#settings` Hash instead:

```ruby
if ENV["USE_SYSTEM_OPENSSL"]
  proj.component "openssl"
  proj.setting(:vendor_openssl, false)
else
 proj.setting(:vendor_openssl, true)
end
```

One of these approaches will probably be more flexible than specifically checking for the String `1` in every component.